azure/services/agentpools/spec.go

```diff
@@ -181,10 +193,33 @@ func (s *AgentPoolSpec) getManagedMachinePoolVersion(existing *asocontainerservi
 }
 
 // Parameters returns the parameters for the agent pool.
-func (s *AgentPoolSpec) Parameters(ctx context.Context, existing *asocontainerservicev1.ManagedClustersAgentPool) (params *asocontainerservicev1.ManagedClustersAgentPool, err error) {
+func (s *AgentPoolSpec) Parameters(ctx context.Context, existingObj genruntime.MetaObject) (params genruntime.MetaObject, err error) {
 	_, _, done := tele.StartSpanWithLogger(ctx, "agentpools.Service.Parameters")
 	defer done()
 
+	// If existing is preview, convert to stable then back to preview at the end of the function.
+	var existing *asocontainerservicev1.ManagedClustersAgentPool
+	var existingStatus asocontainerservicev1preview.ManagedClusters_AgentPool_STATUS
+	if existingObj != nil {
+		if s.Preview {
+			existingPreview := existingObj.(*asocontainerservicev1preview.ManagedClustersAgentPool)
+			existingStatus = existingPreview.Status
+			hub := &asocontainerservicev1hub.ManagedClustersAgentPool{}
+			err := existingPreview.ConvertTo(hub)
```

Comment:
```suggestion
			if err := existingPreview.ConvertTo(hub); err != nil {
```

CAPZ seems to prefer the inline version of Go error checking when possible (optional suggestion of course).